pkg/southbound/gnmi/subscription.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"strings"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+
+	"github.com/onosproject/onos-config/pkg/utils"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// Subscription subscription request information
+type Subscription struct {
+	updatesOnly       bool
+	prefix            string
+	mode              string
+	streamMode        string
+	sampleInterval    uint64
+	heartbeatInterval uint64
```

Comment:
Can we use `time.Duration' for these and convert them internally? It’s impossible to tell what the expected precision of these interval fields even are otherwise. Seconds? Milliseconds? Nanoseconds?

Reply:
The Subscription request itself uses uint64 types not duration. I think we can get back to it later. There is no use of this util function. I added it for future.